projects/components/src/utils/common-util.spec.ts

```diff
@@ -15,4 +16,22 @@ describe('CommonUtil', () => {
     it('can round to 0 when no number is given', () => {
         expect(CommonUtil.roundTo(null)).toEqual(0);
     });
+
+    describe('createBufferedPromise', () => {
+        it(
+            'returns a function that returns a promise which resolves with the output of function passed after a buffer amount of' +
+                'milliseconds',
+            async () => {
+                const bufferedFuncOutput = `output of function passed to createBufferedPromise`;
+                const functionToBeBuffered = () => bufferedFuncOutput;
+                const bufferTime = 400;
+                const bufferedPromise = CommonUtil.createBufferedPromise(functionToBeBuffered, null, bufferTime);
+                let result;
+                setTimeout(() => expect(result).toBeUndefined(), 399);
```

Comment:
Will this test take 400 milliseconds? Can we use fakeAsync to make it not hold up the testing thread for 400 milliseconds?